runner/run.go

```diff
@@ -55,6 +59,14 @@ func Run(ctx context.Context, wg *sync.WaitGroup, globalCollectionOpts state.Col
 				logger.PrintError("Failed to initialize OpenTelemetry tracing provider, disabling exports: %s", err)
 			}
 		}
+
+		if cfg.DbUseIamAuth && cfg.SystemType == "google_cloudsql" && driverCleanup == nil {
```

Comment:
I was initially registering the driver for everyone (as I thought registering won't do any change unless we actually use it), but turned out that it doesn't work and integration tests (aka testing outside of Google Cloud SQL) failed with the following error.

> Failed to register cloudsql-postgres driver: failed to create default credentials: credentials: could not find default credentials. See https://cloud.google.com/docs/authentication/external/set-up-adc for more information

Since there was no problem running this with the GCP VM, I added this condition to only register when it's needed.